listenbrainz/webserver/static/js/src/MissingMBData.tsx

```diff
@@ -0,0 +1,338 @@
+    }
+    this.setState({ loading: false });
+  };
+
+  submitMissingData = (listen: Listen) => {
+    // This function submits data to the MusicBrainz server. We have not used
+    // fetch here because the endpoint where the submision is being done
+    // replies back with HTML and since we cannot redirect via fetch, we have
+    // to resort to such obscure method :D
+    const { user } = this.props;
+    const form = document.createElement("form");
+    form.method = "post";
+    form.action = "https://musicbrainz.org/release/add";
+    form.target = "_blank";
+    const name = document.createElement("input");
+    name.type = "hidden";
+    name.name = "name";
+    name.value = listen.track_metadata.release_name || "";
+    form.appendChild(name);
+    const recording = document.createElement("input");
+    recording.type = "hidden";
+    recording.name = "mediums.0.track.0.name";
+    recording.value = listen.track_metadata.track_name;
+    form.appendChild(recording);
+    const artists = listen.track_metadata.artist_name.split(",");
+    artists.forEach((artist, index) => {
+      const artistCredit = document.createElement("input");
+      artistCredit.type = "hidden";
+      artistCredit.name = `artist_credit.names.${index}.artist.name`;
+      artistCredit.value = artist;
+      form.appendChild(artistCredit);
+      if (index !== artists.length - 1) {
+        const joiner = document.createElement("input");
+        joiner.type = "hidden";
+        joiner.name = `artist_credit.names.${index}.join_phrase`;
+        joiner.value = ", ";
+        form.appendChild(joiner);
+      }
+    });
+    const editNote = document.createElement("textarea");
+    editNote.style.display = "none";
+    editNote.name = "edit_note";
+    editNote.value = `Imported from ${user.name}'s ListenBrainz Missing MusicBrainz Data Page`;
+    form.appendChild(editNote);
+    document.body.appendChild(form);
+    form.submit();
+    form.remove();
+  };
+
+  render() {
+    const {
+      missingData,
+      currPage,
+      totalPages,
+      loading,
+      direction,
+    } = this.state;
+    const { user, newAlert } = this.props;
+    const { APIService } = this.context;
+    return (
+      <div role="main">
+        <div className="row" style={{ display: "flex", flexWrap: "wrap" }}>
+          <div className="col-xs-12">
+            <div>
+              <div id="missingMBData" ref={this.MissingMBDataTable}>
+                <h2>Missing Data:</h2>
+                <div
+                  style={{
+                    height: 0,
+                    position: "sticky",
+                    top: "50%",
+                    zIndex: 1,
+                  }}
+                >
+                  <Loader isLoading={loading} />
+                </div>
+                {missingData.map((data) => {
+                  const listen = {
+                    listened_at:
+                      new Date(`${data.listened_at}Z`).getTime() / 1000,
```

Comment:
Spark gives off timestamp as GMT. So I added a Z at the end to make the browser understand that. Or else it would automatically assume the timestamp to be of local time zone. I infact read up a bit and realised that Date constructor parsing non standard timestamp string should be kind of avoided for consistency reasons (across different browsers). So I think we should discuss this in IRC.